src/openMVG/multiview/solver_resection_p4pf.hpp

```diff
@@ -0,0 +1,127 @@
+ * Copyright (c) 2011, Laurent Kneip, ETH Zurich
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *     * Redistributions of source code must retain the above copyright
+ *       notice, this list of conditions and the following disclaimer.
+ *     * Redistributions in binary form must reproduce the above copyright
+ *       notice, this list of conditions and the following disclaimer in the
+ *       documentation and/or other materials provided with the distribution.
+ *     * Neither the name of ETH Zurich nor the
+ *       names of its contributors may be used to endorse or promote products
+ *       derived from this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL ETH ZURICH BE LIABLE FOR ANY
+ * DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+// Copyright (c) 2012, 2013 Pierre MOULON.
+
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+#ifndef OPENMVG_MULTIVIEW_RESECTION_P4PF_H_
+#define OPENMVG_MULTIVIEW_RESECTION_P4PF_H_
+
+#include <iostream>
+#include "openMVG/numeric/numeric.h"
+
+namespace openMVG {
+namespace resection {
+
+double dabs(double a);
+
+void GJ(double *A, int rcnt, int ccnt, double tol);
+
+// prepare polynomial coefficients
+void CalcCoefs(double const *src1, double const *src2, double const *src3, double const *src4,
+               double const *src5, double *dst1);
+
+// [glab, glac, glad, glbc, glbd, glcd], [a1; a2], [b1; b2], [c1; c2], [d1;d2]
+//  glXY - ||X-Y||^2 - quadratic distances between 3D points X and Y
+//  a1 (a2) = x (resp y) measurement of the first 2D point
+//  b1 (b2) = x (resp y) measurement of the second 2D point
+//  c1 (c2) = x (resp y) measurement of the third 2D point
+//  d1 (d2) = x (resp y) measurement of the fourth 2D point
+//
+// output *A - 10x10 action matrix
+void compute_p4pf_poses(double *glab, double *a1, double *b1, double *c1, double *d1, double *A);
+
+bool isNan(Eigen::MatrixXcd *A);
+
+bool validSol(Eigen::MatrixXcd *sol, Mat *vSol);
+
+void getRigidTransform(Mat *pp1, Mat *pp2, Mat *R, Vec3 *t);
```

Comment:
you can just use the eigen function ::umeyama to register two set of corresponding points
https://eigen.tuxfamily.org/dox/group__Geometry__Module.html#gab3f5a82a24490b936f8694cf8fef8e60
If you don't need the scale factor taken into account, just pass false for the last parameter